src/stateManager/StateManager.ts

```diff
@@ -1308,22 +1313,6 @@ class StateManager {
 
         const previousBlock = previousBlockOrSnapshot.block;
 
-        let previousBlockProducerPostedCalldata = false;
-        if (previousBlock) {
-            if (previousBlock.onChainTimestamp) {
-                previousBlockProducerPostedCalldata = true;
-            } else {
-                previousBlockProducerPostedCalldata = (
-                    await this.diamondStateMachine.localDiamondContract.getBlockCallDataCommitment(
-                        this.channelId,
-                        forkId,
-                        previousBlock.height,
-                        previousBlock.author
-                    )
-                ).found;
```

Comment:
why is this removed

Reply:
see issue description:
> removed the "band-aid" code that was superficially fixing this bug ([link](https://github.com/peer3to/state-channels-plus/commit/3cbe64ea1a1cd1583ccdd9b26c697c02609a5c27#r169464660))

Reply:
honestly i am not sure what you are talking about
this is a new code that i added in the commit i asked you to review in the other PR
it was just masking the issue that onChainTimestamp was undefined

since i have fixed it from the root cause, i have removed this code.
not see what the issue is.
original code, that you have written is

```ts
           previousBlockProducerPostedCalldata: previousBlock
                ? Boolean(previousBlock.onChainTimestamp)
                : false,
```

Reply:
restored and added a test
however, this triggered a new , yet untested, buggy behavior:
- Produce and gossip a correct block N and post some junk as calldata for block N on-chain, 
    * The posting of junk calldata triggers the other peers to create a dispute
    * The uploading of this dispute fails for unknown reason
- The next peer doesn't produce block N+1
- A timeout will be initiated for block N+1 after some time
    * The failure of the dispute above, corrupts something (idk what yet) that causes the timeout dispute to not be created